### PR TITLE
ON-9175 # added user token to request header

### PR DESCRIPTION
**Pull request by simonmarklar**

### CODE REVIEW

File: src/components/LookupNotification.tsx [hunk 1]
```diff
@@ -348,6 +349,12 @@ async function fetchLookup(
     }
   }
 
+  // check for userToken in the query string
+  const qs = querystring.parse(location.search)
+  if (qs.userToken) {
+    headers['X-OneBlink-User-Token'] = qs.userToken as unknown as string
+  }
```
Comment (mymattcarroll) [inline src/components/LookupNotification.tsx:356]: Ideally this comes from the authService as well like the Id token. That way devs can use it without having to know what query string parameters are required to use it.

Comment (simonmarklar) [reply src/components/LookupNotification.tsx:356]: yeah makes sense but auth service comes from apps which has not been updated yet

File: src/components/LookupNotification.tsx [hunk 2]
```diff
@@ -348,6 +348,12 @@ async function fetchLookup(
     }
   }
 
+  // check for userToken in the query string
```
Comment (mymattcarroll) [inline src/components/LookupNotification.tsx:351]: comment is outdated.

File: src/form-elements/FormElementAutocomplete.tsx [hunk 3]
```diff
@@ -1,6 +1,7 @@
 import * as React from 'react'
 import AbortController from 'abort-controller'
 import clsx from 'clsx'
+import querystring from 'query-string'
```
Comment (mymattcarroll) [inline src/form-elements/FormElementAutocomplete.tsx:4]: can we remove this?

File: src/form-elements/FormElementAutocomplete.tsx [hunk 4]
```diff
@@ -162,6 +163,12 @@ const AutocompleteFetch = React.memo(function AutocompleteFetch({
           }
         }
 
+        // check for userToken in the query string
```
Comment (mymattcarroll) [inline src/form-elements/FormElementAutocomplete.tsx:166]: comment is outdated.

**Pull request merged**